commit a8e653b446169b021f31af6e4e6a3494d0f02a8c (refs/changes/61/3561/2)
Author: Jeff Emershaw <jeff.emershaw@joyent.com>
Date:   2018-06-20T16:43:50+02:00 (1 year, 4 months ago)
    
    PUBAPI-1378 Provide UpdateFabricNetwork endpoint
    Reviewed by: Michael Zeller <mike.zeller@joyent.com>

diff --git a/deps/jsstyle b/deps/jsstyle
index d75b7ca..ccb145b 160000
--- a/deps/jsstyle
+++ b/deps/jsstyle
@@ -1 +1 @@
-Subproject commit d75b7ca8308be17c80e2b120f2a01d4a0c20d8a8
+Subproject commit ccb145bab281b81e1fbebd45b5d2c084c454df5d
diff --git a/lib/cloudapi/networks.js b/lib/cloudapi/networks.js
index 4395e54..e346bcc 100644
--- a/lib/cloudapi/networks.js
+++ b/lib/cloudapi/networks.js
@@ -25,6 +25,23 @@ var s_create_network = {
 	'additionalProperties': false
 };
 
+var s_update_network = {
+	'type': 'object',
+	'properties': {
+		'id': base.str.uuid.req,
+		'name': base.str.netname.opt,
+		'routes': base.obj.opt,
+		'description': base.str.opt,
+		'resolvers':
+		    schema_util.array_of(base.str.v4addr.opt),
+		'provision_start_ip': base.str.v4addr.opt,
+		'provision_end_ip': base.str.v4addr.opt,
+		'vlan_id': base.intg.vlanid.req
+	},
+	'additionalProperties': false,
+	'required': [ 'vlan_id', 'id' ]
+};
+
 var s_get_network = {
 	'type': 'object',
 	'properties': {
@@ -47,6 +64,7 @@ var s_list_networks = {
 module.exports = {
     CreateFabricNetwork: s_create_network,
     DeleteFabricNetwork: s_get_network,
+    UpdateFabricNetwork: s_update_network,
     GetFabricNetwork: s_get_network,
     ListFabricNetworks: s_list_networks
 };
diff --git a/test/tst.cloudapi.js b/test/tst.cloudapi.js
index 1fbaec8..3e96d20 100644
--- a/test/tst.cloudapi.js
+++ b/test/tst.cloudapi.js
@@ -60,6 +60,33 @@ test('CreateFabricNetwork', function (t) {
 	t.end();
 });
 
+test('UpdateFabricNetwork', function (t) {
+	var schema = s_cloudapi.UpdateFabricNetwork;
+	var baseParams = {
+		'id': '451640f5-a385-4924-bc3f-a16cf7eacaf2',
+		'name': 'asdf',
+		'vlan_id': 4
+	};
+
+	h.expectValidationErrors(t, schema, {}, [
+		h.errMissing('id'),
+		h.errMissing('vlan_id')
+	]);
+
+	h.expectSingleValidationError(t, schema,
+		extend(baseParams, { 'name': {} }),
+		'name', h.msg.objStr);
+
+	h.expectSingleValidationError(t, schema,
+		extend(baseParams, { 'routes' : 'a' }),
+		'routes', h.msg.strObj);
+
+	h.expectSingleValidationError(t, schema,
+		extend(baseParams, { 'resolvers' : 'a' }),
+		'resolvers', h.msg.strArr);
+
+	t.end();
+});
 
 test('CreateFabricVLAN', function (t) {
 	var schema = s_cloudapi.CreateFabricVLAN;
